File: stars-app/frontend/src/types/member.ts
import type { Channel, GapStatus } from "./common";

export interface MemberGapRow {
  member_key: string;
  display_name: string;
  age: number;
  propensity_score: number;
  measure_code: string;
  measure_name: string;
  gap_status: GapStatus;
  last_contact: string;
  recommended_channel: Channel;
  pcp_name: string;
  campaign_name?: string | null;
}

export interface MemberGapPage {
  items: MemberGapRow[];
  total: number;
  page: number;
  page_size: number;
}

export interface MemberClinicalEvent {
  event_date: string;
  event_type: string;
  description: string;
  provider: string;
}

export interface MemberMedication {
  drug_name: string;
  ndc: string;
  days_supply: number;
  last_fill: string;
  adherent: boolean;
}

export interface MemberProfile {
  member_key: string;
  display_name: string;
  age: number;
  gender: string;
  plan_name: string;
  contract_id: string;
  dual_eligible: boolean;
  lis_flag: boolean;
  utilization_segment: string;
  propensity_score: number;
  open_gaps: MemberGapRow[];
  clinical_events: MemberClinicalEvent[];
  medications: MemberMedication[];
}

File: stars-app/backend/models/member.py
from pydantic import BaseModel
from .common import Channel, GapStatus


class MemberGapRow(BaseModel):
    member_key: str
    display_name: str
    age: int
    propensity_score: float
    measure_code: str
    measure_name: str
    gap_status: GapStatus
    last_contact: str
    recommended_channel: Channel
    pcp_name: str
    campaign_name: str | None = None


class MemberGapPage(BaseModel):
    items: list[MemberGapRow]
    total: int
    page: int
    page_size: int


class MemberClinicalEvent(BaseModel):
    event_date: str
    event_type: str
    description: str
    provider: str


class MemberMedication(BaseModel):
    drug_name: str
    ndc: str
    days_supply: int
    last_fill: str
    adherent: bool


class MemberProfile(BaseModel):
    member_key: str
    display_name: str
    age: int
    gender: str
    plan_name: str
    contract_id: str
    dual_eligible: bool
    lis_flag: bool
    utilization_segment: str
    propensity_score: float
    open_gaps: list[MemberGapRow]
    clinical_events: list[MemberClinicalEvent]
    medications: list[MemberMedication]
